### PR TITLE
LPS-172205 Add API for Testray 2

### PR DESCRIPTION
**Pull request by joelrocha68**
[LPS-172205](https://issues.liferay.com/browse/LPS-172205)

### CODE REVIEW

**Comment by joelrocha68** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayAPI.macro [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+					"r_runToCaseResult_c_runId": "${runId}"
+				}'
+		''';
+
+		var response = JSONCurlUtil.post("${curl}", "$.id");
+
+		echo("${response}");
+
+		return "${response}";
+	}
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayAPI.macro:142]: @joelrocha68, in this case, couldn't we just create 1 macro? The macro is doing the same thing, but just with different casesIDs

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayAPI.macro [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+					"r_taskToSubtasks_c_taskId": "${taskId}"
+				}'
+		''';
+
+		var response = JSONCurlUtil.post("${curl}", "$.id");
+
+		echo("${response}");
+
+		return "${response}";
+	}
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayAPI.macro:523]: Same as comments above.

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayAPI.macro [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+					"r_caseResultToSubtasksCasesResults_c_caseResultId": "${caseResultId}",
+					"r_subtaskToSubtasksCasesResults_c_subtaskId": "${subtaskId}"
+				}'
+		''';
+
+		var response = JSONCurlUtil.post("${curl}", "$.id");
+
+		echo("${response}");
+
+		return "${response}";
```
Comment (barrosdennys) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/macros/TestrayAPI.macro:233]: Same as comment above.

**Comment by luanakarla** [comment]
ci:close

**Pull request closed without merging**